src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/CompiledRegexRunner.cs

```diff
@@ -19,10 +24,11 @@ internal sealed class CompiledRegexRunner : RegexRunner
 
         internal delegate void ScanDelegate(RegexRunner runner, ReadOnlySpan<char> text);
 
-        public CompiledRegexRunner(ScanDelegate scan, CultureInfo? culture)
+        public CompiledRegexRunner(ScanDelegate scan, CultureInfo? culture, MultiStringMatcher? prefixMatcher)
```

Comment:
My comments about the source generator emitting code specific to a trie apply here as well.